### PR TITLE
Adjust filters used in query

### PR DESCRIPTION
**Pull request by tnking97**
This PR includes the following updates:
-Add NIRSpec gratings as options in query form
-Use MIRI subarrays instead of apertures
-Adjust keywords used in query search
-A few adjustments for improved run time
-Reformat html page to allow for varying filters for each instrument
-Include FGS query options
-Adjusted AJAX views to show loading symbol earlier
-Removed unnecessary query filter wildcards to improve run time

### CODE REVIEW

**Comment by tnking97** [comment]
I think it would be best to keep the updated change (where FGS has multiple options for `READPATT_PER_INST` in the `constants.py` file).

On another note, I am having ongoing discussions with the MAST team regarding run time of queries and effects of wildcard parameters; hopefully some of these changes can be implemented shortly.

File: jwql/website/apps/jwql/data_containers.py [hunk 1]
```diff
@@ -887,47 +891,46 @@ def get_thumbnails_all_instruments(parameters):
...
+            if (parameters['detectors'][inst.lower()] != []):
+                query_filters.append({"paramName": "detector", "values": parameters['detectors'][inst.lower()]})
+            if (parameters['filters'][inst.lower()] != []):
+                query_filters.append({"paramName": "filter", "values": parameters['filters'][inst.lower()]})
+            if (parameters['exposure_types'][inst.lower()] != []):
+                query_filters.append({"paramName": "exp_type", "values": parameters['exposure_types'][inst.lower()]})
+            if (parameters['read_patterns'][inst.lower()] != []):
+                query_filters.append({"paramName": "readpatt", "values": parameters['read_patterns'][inst.lower()]})
+            params = {"columns": "*",
+                      "filters": query_filters}
```
Comment (bourque) [inline jwql/website/apps/jwql/data_containers.py:924]: Nice investigative work here to figure out how to get around the wildcard parameter!

**Review by bourque — CHANGES_REQUESTED** [review]
Thank you for all of these changes @tnking97!  They all look really good to me!

I just had some (hopefully very minor) comments/questions, otherwise I think the code changes are good to go.

I still need to test this branch out on the dev server to make sure it works there, but things seem to be working when I test this locally.

File: jwql/utils/utils.py [hunk 2]
```diff
@@ -515,6 +515,42 @@ def get_base_url():
     return base_url
 
 
+def get_instrument_from_filename(filename):
```
[the remark this answers is not in the document]
Comment (tnking97) [reply jwql/utils/utils.py]: Thank you for this idea!

**Comment by tnking97** [comment]
Thank you for your review, @bourque ! I've made updates to address all of the suggestions.

**Review by bourque — APPROVED** [review]
Thanks again @tnking97 for all of these changes!

Upon testing, it looks like this is all working! (albeit slowly; perhaps we can iterate on it in the future to try to improve the speeds further).

**Pull request merged**